libs/ha/ha_common_libs_k8s.py

```diff
@@ -648,21 +651,40 @@ def create_bucket_copy_obj(s3_test_obj=None, bucket_name=None, object_name=None,
                 LOGGER.info("Response: %s", resp)
                 if not resp[0]:
                     return resp if not background else sys.exit(1)
-            status, response = s3_test_obj.copy_object(source_bucket=bucket_name,
-                                                       source_object=object_name,
-                                                       dest_bucket=bkt_name,
-                                                       dest_object=obj_name)
-            LOGGER.info("Response: %s", response)
-            copy_etag = response['CopyObjectResult']['ETag']
-            if put_etag == copy_etag:
-                LOGGER.info("Object %s copied to bucket %s with object name %s successfully",
-                            object_name, bkt_name, obj_name)
-            else:
-                LOGGER.info("Failed to copy object %s to bucket %s with object name %s",
-                            object_name, bkt_name, obj_name)
-                return False, response if not background else sys.exit(1)
 
-        return True, put_etag if not background else output.put((True, put_etag))
+        LOGGER.info("Start copy object to buckets: %s", list(bkt_obj_dict.keys()))
+        for bkt_name, obj_name in bkt_obj_dict.items():
+            try:
+                status, response = s3_test_obj.copy_object(source_bucket=bucket_name,
+                                                           source_object=object_name,
+                                                           dest_bucket=bkt_name,
+                                                           dest_object=obj_name)
+                LOGGER.info("Response: %s", response)
+                copy_etag = response['CopyObjectResult']['ETag']
+                if put_etag == copy_etag:
+                    LOGGER.info("Object %s copied to bucket %s with object name %s successfully",
+                                object_name, bkt_name, obj_name)
+                else:
+                    LOGGER.error("Etags don't match for copy object %s to bucket %s with object "
+                                 "name %s", object_name, bkt_name, obj_name)
+            except CTException as error:
+                LOGGER.error("Error: %s", error)
+                if event.is_set():
+                    exp_fail_bkt_obj_dict[bkt_name] = obj_name
+                    event_clear_flg = True
+                else:
+                    if event_clear_flg:
+                        exp_fail_bkt_obj_dict[bkt_name] = obj_name
+                        event_clear_flg = False
+                        continue
+                    failed_bkts.append(bkt_name)
+                LOGGER.info("Failed to copy object to bucket %s", bkt_name)
+
+        if failed_bkts and not background:
```

Comment:
so in background process we are expecting failures and handling for now at test level with either or?

Reply:
failed_bkts will contain list of bucket with unexpected failures.
For background process we are handling it at test level, same as that of other background methods.
In foreground process, we will return false from here if failed_bkts has buckets in it.